src/Type/Php/ArrayChangeKeyCaseFunctionReturnTypeExtension.php

```diff
@@ -0,0 +1,156 @@
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\TypeTraverser;
+use PHPStan\Type\TypeUtils;
+use PHPStan\Type\UnionType;
+use function array_map;
+use function count;
+use function strtolower;
+use function strtoupper;
+use const CASE_LOWER;
+use const CASE_UPPER;
+
+final class ArrayChangeKeyCaseFunctionReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return $functionReflection->getName() === 'array_change_key_case';
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): ?Type
+	{
+		if (!isset($functionCall->getArgs()[0])) {
+			return null;
+		}
+
+		$arrayType = $scope->getType($functionCall->getArgs()[0]->value);
+		if (!isset($functionCall->getArgs()[1])) {
+			$case = CASE_LOWER;
+		} else {
+			$caseType = $scope->getType($functionCall->getArgs()[1]->value);
+			$scalarValues = $caseType->getConstantScalarValues();
+			if (count($scalarValues) === 1) {
+				$case = (int) $scalarValues[0];
+			} else {
+				$case = null;
+			}
+		}
+
+		$constantArrays = $arrayType->getConstantArrays();
+		if (count($constantArrays) > 0) {
+			$arrayTypes = [];
+			foreach ($constantArrays as $constantArray) {
+				$newConstantArrayBuilder = ConstantArrayTypeBuilder::createEmpty();
+				foreach ($constantArray->getKeyTypes() as $i => $keyType) {
+					$valueType = $constantArray->getOffsetValueType($keyType);
+
+					$constantStrings = $keyType->getConstantStrings();
+					if (count($constantStrings) > 0) {
+						$keyType = TypeCombinator::union(
+							...array_map(
+								fn (ConstantStringType $type): Type => $this->mapConstantString($type, $case),
+								$constantStrings,
+							),
+						);
+					}
+
+					$newConstantArrayBuilder->setOffsetValueType(
+						$keyType,
+						$valueType,
+						$constantArray->isOptionalKey($i),
+					);
+				}
+				$newConstantArrayType = $newConstantArrayBuilder->getArray();
+				if ($constantArray->isList()->yes()) {
+					$newConstantArrayType = AccessoryArrayListType::intersectWith($newConstantArrayType);
+				}
+				$arrayTypes[] = $newConstantArrayType;
+			}
+
+			$newArrayType = TypeCombinator::union(...$arrayTypes);
+		} else {
+			$keysType = $arrayType->getIterableKeyType();
+
+			$keysType = TypeTraverser::map($keysType, function (Type $type, callable $traverse) use ($case): Type {
+				if ($type instanceof UnionType) {
+					return $traverse($type);
+				}
+
+				$constantStrings = $type->getConstantStrings();
+				if (count($constantStrings) > 0) {
+					return TypeCombinator::union(
+						...array_map(
+							fn (ConstantStringType $type): Type => $this->mapConstantString($type, $case),
+							$constantStrings,
+						),
+					);
+				}
+
+				if ($type->isString()->yes()) {
+					if ($case === CASE_LOWER) {
+						return TypeCombinator::intersect($type, new AccessoryLowercaseStringType());
```

Comment:
Lower can still be combined with non-empty/non-false ?

Reply:
It was already the case because I intersect.
But I rework this because I also need to lose the literal Accessory.